### PR TITLE
fix: fix session proto open/close by user part

### PR DESCRIPTION
**Pull request by driftluo**
This is a long-established bug that only affects the manual opening of the user's protocol at the session-level.

There's a logical confusion here, the handle doesn't need to be open every time it's open, because the session is already resident when it's open. The handle does not need to be removed when the protocol is closed alone, but only when the session is closed.

Now, with the addition of a test case here.

### CODE REVIEW

File: tests/test_session_handle_open.rs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+    let meta_dummy_2 = MetaBuilder::new()
+        .id(0.into())
+        .service_handle(move || {
+            let handle = Box::new(Dummy);
+            ProtocolHandle::Callback(handle)
+        })
+        .build();
+
+    let meta_2 = MetaBuilder::new()
```
Comment (quake) [inline tests/test_session_handle_open.rs:99]:
```suggestion
    let meta_1 = MetaBuilder::new()
```

File: tests/test_session_handle_open.rs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        .id(1.into())
+        .session_handle(move || {
+            let handle = Box::new(p_handle_1.clone());
+            ProtocolHandle::Callback(handle)
+        })
+        .build();
+
+    let meta_2_1 = MetaBuilder::new()
```
Comment (quake) [inline tests/test_session_handle_open.rs:107]:
```suggestion
    let meta_2 = MetaBuilder::new()
```

File: tests/test_session_handle_open.rs [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                .unwrap();
+
+            addr_sender.send(listen_addr).unwrap();
+
+            loop {
+                if service_2.next().await.is_none() {
+                    break;
+                }
+            }
+        });
```
Comment (quake) [inline tests/test_session_handle_open.rs:134]: suggest adding `assert(10, count)`

Comment (driftluo) [reply tests/test_session_handle_open.rs:134]: You cannot confirm which service has reached 10 unless you confirm the direction of the dial

Comment (TheWaWaR) [reply tests/test_session_handle_open.rs:134]: `service_1` dial `service_2`, so `service_1` is inbound to `service_2`.

Comment (driftluo) [reply tests/test_session_handle_open.rs:134]: > `service_1` dial `service_2`, so `service_1` is inbound to `service_2`.

But in subsequent dials, it is whoever receives the shutdown task that tries to dial to the other

Comment (TheWaWaR) [reply tests/test_session_handle_open.rs:134]: You can assert the sum

Comment (driftluo) [reply tests/test_session_handle_open.rs:134]: I don't think that's the point, without reach to 10, the service can't automatically shutdown

**Pull request merged**